no such file or directory, open '/path/schema.prisma' since 3.1.1

### Bug description

When I call ` new PrismaClient()`, I get the error 
```bash
(node:12802) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/mnt/d/workspace/path/packages/api/src/schema.prisma'
```
That just happen on 3.1.1

### How to reproduce

1. Upgrade prisma to 3.1.1
2. Execute `prisma db push` command
3. Call ` new PrismaClient()`

### Expected behavior

_No response_

### Prisma information

<!-- Do not include your database credentials when sharing your Prisma schema! -->


### Environment & setup

- OS: Windows(wsl)
- Database: SQLite
- Node.js version: v14.17.4


### Prisma Version

```
prisma                  : 3.1.1
@prisma/client          : 3.1.1
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine c22652b7e418506fab23052d569b85d3aec4883f (at ../../.yarn/unplugged/@prisma-engines-npm-3.1.0-24.c22652b7e418506fab23052d569b85d3aec4883f-4e2ba773cc/node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli c22652b7e418506fab23052d569b85d3aec4883f (at ../../.yarn/unplugged/@prisma-engines-npm-3.1.0-24.c22652b7e418506fab23052d569b85d3aec4883f-4e2ba773cc/node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core c22652b7e418506fab23052d569b85d3aec4883f (at ../../.yarn/unplugged/@prisma-engines-npm-3.1.0-24.c22652b7e418506fab23052d569b85d3aec4883f-4e2ba773cc/node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt c22652b7e418506fab23052d569b85d3aec4883f (at ../../.yarn/unplugged/@prisma-engines-npm-3.1.0-24.c22652b7e418506fab23052d569b85d3aec4883f-4e2ba773cc/node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : c22652b7e418506fab23052d569b85d3aec4883f
Studio                  : 0.423.0
```

@andyjy are you able to share a sample project that includes the different pieces?

I have this in a `db` package:
```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
  generator = "../../../node_modules/@~internal/prisma-myschema/client"
}
```

And when I import the following into my `web` package (a Next.js project). I get the following error:

```ts
import { PrismaClient } from "@~internal/prisma-myschema/client"
```

![image](https://user-images.githubusercontent.com/3090527/142733335-e054680a-d979-46a9-be04-e830d2dbcc43.png)

Or in Next:

```
Module not found: Can't resolve '@~internal/prisma-myschema/client'
```

--- a/packages/client/src/generation/utils/buildDirname.ts
+++ b/packages/client/src/generation/utils/buildDirname.ts
@@ -24,20 +24,20 @@ export function buildDirname(clientEngineType: ClientEngineType, relativeOutdir:
  * feature is especially useful for Next.js/Webpack users since the client gets
  * moved and copied out of its original spot. It all fails, it falls-back to
  * `__dirname`, which is never available on bundles.
- * @param relativeOutdir
+ * @param defaultRelativeOutdir
  * @param runtimePath
  * @returns
  */
-function buildDirnameFind(relativeOutdir: string, runtimePath: string) {
+function buildDirnameFind(defaultRelativeOutdir: string, runtimePath: string) {
   // potential client location on serverless envs
-  const slsRelativeOutputDir = relativeOutdir.split(path.sep).slice(1).join(path.sep)
+  const serverlessRelativeOutdir = defaultRelativeOutdir.split(path.sep).slice(1).join(path.sep)
 
   return `
 const { findSync } = require('${runtimePath}')
 
 const dirname = findSync(process.cwd(), [
-    ${JSON.stringify(relativeOutdir)},
-    ${JSON.stringify(slsRelativeOutputDir)},
+    ${defaultRelativeOutdir ? `${JSON.stringify(defaultRelativeOutdir)},` : ''}
+    ${serverlessRelativeOutdir ? `${JSON.stringify(serverlessRelativeOutdir)},` : ''}
 ], ['d'], ['d'], 1)[0] || __dirname`
 }
 

